packages/ui-core/src/components/Search/Search.docz.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+interface IDemoSearchWrapperProps {
+    children: (prop: {
+        onSubmit: (value: string) => void;
+    }) => JSX.Element;
+}
+
+export const DemoSearchWrapper: React.FC<IDemoSearchWrapperProps> = ({ children }) => {
+    const [value, setValue] = React.useState('');
+
+    const handleSubmit = (chosenValue: string): void => {
+        setValue(chosenValue);
+    };
+
+    return (
+        <div>
+            <p style={{ marginTop: 0 }}>Value: {value}</p>
+            {children({ onSubmit: handleSubmit })}
```

Comment:
```suggestion
            {children({ onSubmit: setValue })}
```
И метод handleSubmit выше вообще не нужен.